src/stylize.ts

```diff
@@ -0,0 +1,98 @@
+import {Channel} from 'vega-lite/src/channel';
+import {Type} from 'vega-lite/src/type';
+
+import {QueryConfig} from './config';
+import {isEnumSpec} from './enumspec';
+import {SpecQueryModel} from './model';
+import {EncodingQuery, ScaleQuery} from './query/encoding';
+import {Schema} from './schema';
+import {Dict} from './util';
+
+export function stylize(answerSet: SpecQueryModel[], schema: Schema, opt: QueryConfig): SpecQueryModel[] {
+
+  answerSet = answerSet.map(function(specM) {
+
+    if (opt.smallBandSizeForHighCardinalityOrFacet) {
+      specM = smallBandSizeForHighCardinalityOrFacet(specM, schema);
+     }
+
+    if (opt.nominalColorScaleForHighCardinality) {
+      specM = nominalColorScaleForHighCardinality(specM, schema);
+    }
+    return specM;
+  });
+
+  return answerSet;
+}
+
+let encQIndex: Dict<EncodingQuery> = {};
+
+export function smallBandSizeForHighCardinalityOrFacet(specM: SpecQueryModel, schema: Schema): SpecQueryModel {
+ [Channel.ROW, Channel.Y, Channel.COLUMN, Channel.X].forEach((channel) => {
+        encQIndex[channel] = specM.getEncodingQueryByChannel(channel);
+      });
+
+  const yEncQ = encQIndex[Channel.Y];
+  if (yEncQ !== undefined) {
+    if (encQIndex[Channel.ROW] ||
+        schema.cardinality(yEncQ) > 10) {
+
+      // We check for undefined rather than
+      // yEncQ.scale = yEncQ.scale || {} to cover the case where
+      // yEncQ.scale has been set to false/null.
+      // This prevents us from incorrectly overriding scale and
+      // assigning a bandSize when scale is set to false
+      if (yEncQ.scale === undefined) {
+        yEncQ.scale = {};
+      }
+
+      // We do not want to assign a bandSize if scale is set to false
+      if (yEncQ.scale && !isEnumSpec(yEncQ.scale)) {
```

Comment:
If scale is enumSpec, but bandSize is undefined, I think it's fine to set `bandSize`.